### ISSUE TITLE
Not Working on lolipop

### ISSUE THREAD

**Issue by masoud-c** [dteviot/epubWorkspace#6]
it seems it has a problem with lolipop devices, it crashes on unzipping part

**Comment by dteviot** [comment]
That's very odd, would suggest there's some issue with backwards compatibility in Lollipop.
I'm not currently set up for building or testing on Lollipop.

The fact that you know it's crashing on the unzipping part suggests you have logs or similar.
If you send me the following:
1. The e-pub file that is crashing.
2. The log file

I'll have a look at it and see what can be done.

**Comment by masoud-c** [comment]
I found the solution:
in this part -> mZip = new ZipFile(fileName)
lollipop device threw an  exception of "Not a zip file", I dont know why
and then a searched a little and find a library called <a href="http://www.lingala.net/zip4j/" target="_blank">Zip4j</a> and then replaced it with ZipFile and now it works on all devices perfectly and it's much faster now
thanks for the answer

**Comment by dteviot** [comment]
I would still like to see the epub file that gave the problem.
I'd like to find what the root problem is.

**Comment by masoud-c** [comment]
it's the epub file:
https://www.dropbox.com/s/84k690vlle6k7fz/5.epub?dl=0

**Comment by dteviot** [comment]
Got it.  Thank you.

**Comment by dteviot** [comment]
What version of Lollipop did you try?
It works fine with the Android 5.0.1 emulator provided by Google.
Although I did have to add 

``` xml
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" /> 
```

to 
AndroidManifest.xml so that the App could see and read the files in the Download directory.
And I also had to remove the WebServer because Google has removed most of the org.apache.http components.  (Probably a way to compile them in, I just didn't have time to figure it out.)

Looks like its add this to the "build.gradle" file
android {
    useLibrary 'org.apache.http.legacy'
}
